_posts/2023-05-25-10block-old-decoy-selection-bug.md

```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: Post-Mortem of 10-Block-Old Decoy Selection Bug
+summary: Patched in official wallet, update highly recommended.
+tags: [core, crypto, community]
+author: jeffro256
+---
+
+## Quick Facts
+
+* Severity: HIGH
+* Affected versions: GUI/CLI wallet versions v0.14.1.0 to v0.18.2.1
+* Impact: Loss of sender anonymity for transactions spending funds exactly 10 blocks old
+* What to do: [Update to v0.18.2.2 and above]({{ site.baseurl_root }}/downloads)
+
+## Introduction
+
+When the Monero wallet needs to construct a @transaction with a certain input, it also picks decoys inputs from the chain (@ring-signature) based on a certain distribution called a "gamma distribution". The gamma distribution makes picking recent decoys more likely than picking older decoys, mimicking how real users spend their funds. However, in the gamma picker code, there was a off-by-one bug that didn't allow the gamma picker to pick decoys which are exactly 10 blocks old.    
+
+The wallet can, however, still spend owned outputs that are exactly 10 blocks old. This means that an external observer can guess the true spend of an input ring with very high likelihood if one of the ring members is exactly 10 blocks old.    
+
+This bug has been patched in v0.18.2.2 and it is recommended that all users update their wallets as soon as possible. Since many third-party wallet applications rely on the core "wallet2" code, if you do not use the Monero Core CLI/GUI wallets, ask the development team of your wallet if they have upgraded to the new wallet2 code.    
+
+Upgrading your wallet to v0.18.2.2 will not only improve your sender anonymity, but will increase the anonymity pool for all other users, including those using the older vulnerable wallet code. If you must absolutely stay behind with an older wallet version, do not ever spend funds as soon as they are unlocked; make sure to wait one extra block before sending.
```

Comment:
> make sure to wait one extra block before sending

If this recommendation ends up causing people to spike pay in 11 blocks versus 10, then that in and of itself could theoretically leave another way to stick out. Like you noted in your prior sentence, since newer wallets can now select 10 block decoys, old wallets are now protected if they spend an output 10 blocks old. I think it's arguably safer not to recommend older wallet users wait 11 blocks and just recommend people update wallets asap.

Reply:
cc @jeffro256 would removing the entire " If you must absolutely stay behind with an older wallet version, do not ever spend funds as soon as they are unlocked; make sure to wait one extra block before sending." sentence be an easy option here?

Reply:
Thanks, that's a good point: if everyone waits 1 block then 11 becomes the new 10. And waiting doesn't change the fact the wallet still won't pick 10-old decoys with an 11-old true spend. I removed that suggestion from the original post.